adafruit_ads1x15/ads1x15.py

```diff
@@ -317,3 +389,22 @@ def _read_register(self, reg: int, fast: bool = False) -> int:
             else:
                 i2c.write_then_readinto(bytearray([reg]), self.buf, in_end=2)
         return self.buf[0] << 8 | self.buf[1]
+
+    def write_config(self, pin_config: int) -> None:
```

Comment:
I can make write_config a private function that is called by every setter and by the read function, but not accessible directly to the outside user. But I obviously cannot just remove "write_config" without just copying that code back in at all the places it is used.

In order to be consistent, I looked through all the other Adafruit ADC drivers and I found the MCP3421 to be the most similar to the ADS1x15 because it also has several parameters that are set through a configuration register. In that driver, every setter calls the "_register_value" function that assembles the configuration bytes and then sends that byte to the ADC through I2C. I just consolidated these steps into a "write_config" function. MCP3421 also has a function to read the configuration register called "_update_config", which is functionally very similar to my "read_config" function.

I agree with you that the driver state should be the device state, but I explained earlier that I ran into a situation where that was not true with the code the way it was. Making these functions more explicit makes it easier to ensure that the driver state is the device state.

Reply:
Making them private with a `_` prefix to the name is fine with me! Sorry I didn't suggest that.